lavalink/AudioTrack.py

```diff
@@ -3,32 +3,48 @@ class InvalidTrack(Exception):
     pass
 
 
+class TrackNotBuilt(Exception):
+    """ This exception will be raised when AudioTrack objects hasn't been built. """
+    pass
+
+
 class AudioTrack:
-    def build(self, track, requester, **kwargs):
+    __slots__ = ("track", "identifier", "can_seek", "author", "duration", "stream", "title", "uri", "requester",
+                 "preferences")
+
+    def __init__(self, requester, **kwargs):
+        self.requester = requester
+        self.preferences = kwargs
+
+    @classmethod
+    def build(cls, track, requester, **kwargs):
         """ Returns an optional AudioTrack. """
+        new_track = cls(requester, **kwargs)
         try:
-            self.track = track['track']
-            self.identifier = track['info']['identifier']
-            self.can_seek = track['info']['isSeekable']
-            self.author = track['info']['author']
-            self.duration = track['info']['length']
-            self.stream = track['info']['isStream']
-            self.title = track['info']['title']
-            self.uri = track['info']['uri']
-            self.requester = requester
-            self.preferences = kwargs
-
-            return self
+            new_track.track = track['track']
+            new_track.identifier = track['info']['identifier']
+            new_track.can_seek = track['info']['isSeekable']
+            new_track.author = track['info']['author']
+            new_track.duration = track['info']['length']
+            new_track.stream = track['info']['isStream']
+            new_track.title = track['info']['title']
+            new_track.uri = track['info']['uri']
+
+            return new_track
         except KeyError:
             raise InvalidTrack('An invalid track was passed.')
 
     @property
     def thumbnail(self):
         """ Returns the video thumbnail. Could be an empty string. """
+        if not hasattr(self, "track"):

```

Comment:
Single quotes

Reply:
I think we should add a double quotes test in `run_tests.py` : ^)

Reply:
If you can figure out how, then go ahead